usr/src/cmd/rsrvrctl/rsrvrctl.c

```diff
@@ -20,10 +21,24 @@
 #include <assert.h>
 #include <signal.h>
 #include <sys/types.h>
+#include <limits.h>
 
 #include <sys/vmm_dev.h>
 
 const char *prog_name;
+#define	LOCK_FILE	"/var/run/rsrvrctl.lock"
+
+static void
+get_lock(void)
+{
+	int fd;
+
+	fd = open(LOCK_FILE, O_CREAT | O_WRONLY, 0600);
+	if (fd == -1)
+		err(EXIT_FAILURE, "Could not open lock file '%s'", LOCK_FILE);
+	if (lockf(fd, F_LOCK, 0) != 0)
+		err(EXIT_FAILURE, "Could not acquire lock");
+}
```

Comment:
Perhaps save the output, how is this any different from the situation we have today?  Or is that covered in, "does not address the problems with concurrency"?

(I have thoughts on that...)